src/renderer/util/index.ts

```diff
@@ -41,26 +62,66 @@ export async function determineFilesFromNestedPaths(
 ): Promise<string[]> {
   const filePaths = await Promise.all(
     paths.flatMap(async (fullPath) => {
-      const stats = await fsPromises.stat(fullPath);
-      if (!stats.isDirectory()) {
-        return [fullPath];
-      }
-      const canRead = await canUserRead(fullPath);
-      if (!canRead) {
-        throw new Error(`User does not have permission to read ${fullPath}`);
-      }
-      const pathsUnderFolder = await fsPromises.readdir(fullPath, {
-        withFileTypes: true,
-      });
-      return pathsUnderFolder
-        .filter((f) => f.isFile())
-        .map((f) => resolve(fullPath, f.name));
+      return determineFilesFromNestedPath(fullPath);
     })
   );
 
   return uniq(flatten(filePaths));
 }
 
+/**
+ * Use a given filepath's extension to determine if it is a "multifile".
+ * @param filePath Path to the file
+ */
+export function determineIsMultifile(filePath: string): boolean {
+  ///////////////////////////////////////////////////////////////////
+  // TODO - Remove this once multifile support is feature-complete //
+  //         Flip the boolean to "true" for testing                //
+  ///////////////////////////////////////////////////////////////////
+  const USE_MULTIFILE_ASSUMPTION = process.env.USE_MULTIFILE_ASSUMPTION !== undefined
+      ? process.env.USE_MULTIFILE_ASSUMPTION === 'true'
+      : false;
+  ///////////////////////////////////////////////////////////////////
+
+  const multifileExtensions = ['.zarr', '.sldy'];
+  const combinedExtensions = multifileExtensions.join('|');
+
+  // "ends with one of the listed extensions, ignoring casing"
+  // otherwise written like: /.zarr|.sldy$/i
+  const matcher = new RegExp(combinedExtensions + "$", 'i');
+
+  // If the regex matches it will return an array (truthy).
+  // If the regex doesn't match it will return null (falsy).
+  return Boolean(filePath.match(matcher)) && USE_MULTIFILE_ASSUMPTION;
+}
+
+/**
+ * Returns the total size of a given directory's children, sub-children, etc.
+ * If the given path points to a file rather than a directory, returns the size of the file.
+ * @param dir Local path to a directory.
+ *
+ * Borrowed from StackOverflow: https://stackoverflow.com/a/69418940
+ */
+export async function getDirectorySize(dir: string): Promise<number> {
+  const files = await readdir(dir, { withFileTypes: true });
+
+  const paths = files.map(async file => {
+    const path = join(dir, file.name);
+
+    if (file.isDirectory()) {
+      return await getDirectorySize(path);
```

Comment:
What about the other contents of `dir`?  Won't this result in just returning the size of the first subdir encountered?  It seems like it should be creating a list of all the file sizes and recursive calls, and summing it all.